apps/web-giddh/src/app/permissions/components/details/permission.details.html

```diff
@@ -27,103 +33,87 @@
-                                                <input
-                                                    id="{{item.code}}+{{idx}}+{{i}}"
-                                                    [disabled]="newRole.isFixed"
-                                                    (change)="toggleItem(page.name, item, $event)"
-                                                    type="checkbox"
-                                                    name="{{item.code}}_{{i}}"
-                                                    [(ngModel)]="item.isSelected"
-                                                />
-                                                <label for="{{item.code}}+{{idx}}+{{i}}">
-                                                    {{getNameByCode(item.code)}}</label
-                                                >
-                                            </td>
-                                            <td class="text-center">
-                                                <i
-                                                    class="fa"
-                                                    [ngClass]="isHavePermission(page.name, item, 'admin') ? 'fa-check' : 'fa-times'"
-                                                    aria-hidden="true"
-                                                ></i>
-                                            </td>
-                                            <td class="text-center">
-                                                <i
-                                                    class="fa"
-                                                    [ngClass]="isHavePermission(page.name, item, 'view') ? 'fa-check' : 'fa-times'"
-                                                    aria-hidden="true"
-                                                ></i>
-                                            </td>
-                                        </ng-container>
-                                    </tr>
-                                </tbody>
-                            </table>
-                        </div>
+                    <div>
+                        <h2 class="pd-l05">{{ page.name }}</h2>
+                        <a
+                            class="close pull-right"
+                            href="javascript:void(0);"
+                            (click)="removePageFromScope(page.name)"
+                            *ngIf="!newRole.isFixed"
+                            >×</a
+                        >
                     </div>
+                    <table mat-table [dataSource]="page.permissions" class="w-100 mr-b4">
+                        <ng-container matColumnDef="admin">
+                            <th mat-header-cell *matHeaderCellDef>{{ newRole.name }}</th>
+                            <td mat-cell *matCellDef="let element, let i = index">
+                                <ng-container *ngIf="element.code === 'SELECT-ALL'">
```

Comment:
```suggestion
                                <ng-container *ngIf="element?.code === 'SELECT-ALL'">
```
check element and page in overall file